vispy/util/frozen.py

```diff
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2015, Vispy Development Team.
+# Distributed under the (new) BSD License. See LICENSE.txt for more info.
+
+
+# Class adapted from:
+# http://stackoverflow.com/questions/3603502/
+#        prevent-creating-new-attributes-outside-init
+
+class Frozen(object):
+    __isfrozen = False
+
+    def __setattr__(self, key, value):
+        if self.__isfrozen and not hasattr(self, key):
+            raise AttributeError("%r is not an attribute of %s"
```

Comment:
I would add `"call unfreeze() to allow addition of new attributes"`  or maybe `"see vispy.util.Frozen"`.